packages/common-all/src/types/store.ts

```diff
@@ -3,10 +3,20 @@ import { DVault } from "./workspace";
 
 // Types used on the store layer
 
+/**
+ * Properties used to find notes by. If multiple properties are provided, then returned notes
+ * must satisfy all constraints.
+ *
+ * For example, if fname = "foo" and vault = "vaultOne", then only notes with fname "foo" in vault "vaultOne" are returned
+ */
 export type FindNoteOpts = {
+  // Find notes by fname
   fname?: string;
   // If vault is provided, filter results so that only notes with matching vault is returned
   vault?: DVault;
+  // Find note by child. This usually refers to parent but could also be closest ancestor if parent doesn't exist. Max one note should be returned
+  // E.g. if childFname = `baz.one.two` and `baz.one` does not exist, then `baz` will be returned
+  child?: NotePropsMeta; // Noteprops metadata of child
```

Comment:
can we rename this to `ancestor` to make this explicit?

Reply:
So I've intended these to be properties we find notes "by". Ie find notes where fname = "foo" or find notes where child = "baz".

For the child property, it technically should return just the parent note, but I've expanded it to include ancestor since we have multiple use cases for it.

I'd like to keep it child to keep it consistent with the other properties

Reply:
Why would someone want to do this? Why not just find the child note explicitly, then get its parent?  

Not sure that this parameter is necessary, but if it is I'd prefer to break it out separately.  It's not clear what it means to "find a note by child".  It'd be better to have a method `getAncestor(id: string)`

Reply:
this is used internally for cases like delete. see https://github.com/dendronhq/dendron/pull/3284/files#diff-ef9b24e9ac675f8a705cd6fd0c0ce40ab1ba6419dd498e51c66d23ba9bc53de9R376

Reply:
But why can't we use parent pointers?

Reply:
Actually the main usage right now is during initNotes or writeNote for when we are adding a new note that doesn't have a parent yet. Ie we want to add `foo.bar.baz` in the engine but only `foo` exists, in which case we need to create a stub for `foo.bar` first.

Reply:
Ah ok - that makes sense.  I still strongly prefer if we either break this out or use different logic to to obtain the parent. 

If `findNotes` becomes the general 'query/SQL-like' API, then it's quickly going to get messy and contradictory.  For example, the natural thing someone else may then add is a 'parent' field - now what do you do if someone sets both arguments? 

As soon as we move beyond name + vault, I think it's gonna be a slippery slope down this path.  name + vault makes sense as parameters since we're saying that it's kind of like a composite primary key for a note.

Reply:
Originally I was treating `findNotes` like a query method. So if parent and child was both set, it'd return all notes where child = foo and parent = bar (ie bar.baz.foo).

But since the ancestor search is specific to the engine, I can move it to a separate logic that doesn't use find

Reply:
I think this makes sense as long as we treat the arguments of `findNotes()` as query parameters on the props of the particular note.  But since this particular use case does special ancestral logic, I think it should be named accordingly.